pkg/controllers/netpol/pod.go

```diff
@@ -99,11 +99,14 @@ func (npc *NetworkPolicyController) syncPodFirewallChains(networkPoliciesInfo []
 	}
 
 	// loop through the pods running on the node
-	allLocalPods, err := npc.getLocalPods(npc.nodeIP.String())
-	if err != nil {
-		return nil, err
+	allLocalPods := make(map[string]podInfo)
+	if npc.enableIPv4 {
+		npc.getLocalPods(allLocalPods, npc.nodeIPv4.String())
 	}
-	for _, pod := range *allLocalPods {
+	if npc.enableIPv6 {
+		npc.getLocalPods(allLocalPods, npc.nodeIPv6.String())
```

Comment:
IIUC, you are not returning any value in the npc.getLocalPods function because you expect the variable allLocalPods to change. Shouldn't it be a pointer then?

Reply:
Maps and slices are reference types in Go, you don't have to use pointers for them

https://go.dev/blog/maps